cnd/src/storage.rs

```diff
@@ -355,16 +355,20 @@ impl From<SwapContextRow> for SwapContext {
 #[async_trait::async_trait]
 impl Load<SwapContext> for Storage {
     async fn load(&self, swap_id: LocalSwapId) -> anyhow::Result<SwapContext> {
+        use self::schema::swap_contexts;
+
         let context = self
             .db
-            .swap_contexts()
+            .do_in_transaction(|connection| {
+                swap_contexts::table
+                    .filter(swap_contexts::local_swap_id.eq(Text(swap_id)))
+                    .get_result::<tables::SwapContext>(connection)
+                    .optional()
```

Comment:
❓ Why do we need this `optional()` here?

Reply:
Because it could not exist potentially so I do `ok_or` a couple of lines down.